lib/vrl/parser/src/ast.rs

```diff
@@ -758,18 +786,38 @@ impl fmt::Debug for Assignment {
         use Assignment::*;
 
         match self {
-            Single { target, expr } => write!(f, "{:?} = {:?}", target, expr),
-            Infallible { ok, err, expr } => write!(f, "Ok({:?}), Err({:?}) = {:?}", ok, err, expr),
+            Single { target, op, expr } => write!(f, "{:?} {} {:?}", target, op, expr),
+            Infallible { ok, err, op, expr } => {
+                write!(f, "Ok({:?}), Err({:?}) {} {:?}", ok, err, op, expr)
+            }
         }
     }
 }
 
-#[derive(PartialEq)]
+#[derive(Clone, PartialEq)]
 pub enum AssignmentTarget {
     Noop,
     Query(Query),
     Internal(Ident, Option<Path>),
-    External(Option<Path>),
+}
+
+impl AssignmentTarget {
+    pub fn into_expr(self, span: &Span) -> Expr {
+        match self {
+            AssignmentTarget::Noop => Expr::Literal(Node::new(span.clone(), Literal::Null)),
+            AssignmentTarget::Query(query) => Expr::Query(Node::new(span.clone(), query)),
+            AssignmentTarget::Internal(ident, Some(path)) => Expr::Query(Node::new(
+                span.clone(),
+                Query {
+                    target: Node::new(span.clone(), QueryTarget::Internal(ident)),
+                    path: Node::new(span.clone(), path),
+                },
+            )),
+            AssignmentTarget::Internal(ident, None) => {
+                Expr::Variable(Node::new(span.clone(), ident))
+            }
```

Comment:
As discussed on Discord. I prefer we keep using `Internal` and `External` for now, even though some recent parser changes have made them obsolete. We should remove them, but I'd rather do that in a follow-up PR that looks at the bigger picture and looks into simplifying the parser as well, now that these types are no longer needed.

(we can create an issue to track this work)